skyportal/facility_apis/atlas.py

```diff
@@ -0,0 +1,272 @@
+            .filter(FollowupRequest.id == request.id)
+            .first()
+        )
+
+        altdata = request.allocation.altdata
+
+        if not altdata:
+            raise ValueError('Missing allocation information.')
+
+        content = req.transactions[0].response["content"]
+        content = json.loads(content)
+
+        r = requests.get(
+            content["url"],
+            headers={
+                'Authorization': f"Token {altdata['api_token']}",
+                'Accept': 'application/json',
+            },
+        )
+        r.raise_for_status()
+
+        if r.status_code == 200:
+            if r.json()['finishtimestamp']:
+                result_url = r.json()['result_url']
+                request.status = (
+                    f"Task is complete with results available at {result_url}"
+                )
+
+                s = requests.get(
+                    result_url,
+                    headers={
+                        'Authorization': f"Token {altdata['api_token']}",
+                        'Accept': 'application/json',
+                    },
+                )
+                s.raise_for_status()
+                df = pd.read_csv(
+                    StringIO(s.text.replace("###", "")), delim_whitespace=True
+                )
+                df.rename(
+                    columns={
+                        'MJD': 'mjd',
+                        'RA': 'ra',
+                        'Dec': 'dec',
+                        'm': 'mag',
+                        'dm': 'magerr',
+                        'mag5sig': 'limiting_mag',
+                        'F': 'filter',
+                    },
+                    inplace=True,
+                )
+                cyan = df['filter'] == 'c'
+                orange = df['filter'] == 'o'
+                snr = df['uJy'] / df['duJy'] < 5
+
+                df['filter'].loc[cyan] = 'sdssg'
+                df['filter'].loc[orange] = 'sdssr'
+                df['mag'].loc[snr] = None
+                df['magerr'].loc[snr] = None
+                df = df.replace({np.nan: None})
+
+                df.drop(
+                    columns=[
+                        'min',
+                        'phi',
+                        'uJy',
+                        'chi/N',
+                        'duJy',
+                        'maj',
+                        'Obs',
+                        'apfit',
+                        'x',
+                        'y',
+                        'Sky',
+                        'err',
+                    ],
+                    inplace=True,
+                )
+                df['magsys'] = 'ab'
+
+                data_out = {
+                    'obj_id': request.obj_id,
+                    'instrument_id': instrument.id,
+                    'group_ids': 'all',
+                    **df.to_dict(orient='list'),
+                }
+                token_id = "720504e1-f163-478b-b624-a96573a68cbb"
```

Comment:
what is this? Should it be in a config? in .secrets?

Reply:
This was just my local token when testing. I assume there is a way to talk to the API directly from within the app without any tokens, but I don't know what that is!

Reply:
I dont think this is our preferred access pattern (APIs calling APIs). @stefanv @guynir42 ?